src/HotChocolate/Language/src/Language.Visitors/SyntaxVisitor~1.cs

```diff
@@ -0,0 +1,139 @@
+using System.Collections.Generic;
+
+namespace HotChocolate.Language.Visitors
+{
+    public class SyntaxVisitor<TContext>
+        : ISyntaxVisitor<TContext>
+        where TContext : ISyntaxVisitorContext
+    {
+        private static readonly SyntaxNodeListPool _listPool = new SyntaxNodeListPool();
+        private static readonly List<ISyntaxNode> _empty = new List<ISyntaxNode>();
+
+        public SyntaxVisitor()
+        {
+            DefaultAction = Skip;
+        }
+
+        public SyntaxVisitor(ISyntaxVisitorAction defaultResult)
+        {
+            DefaultAction = defaultResult;
+        }
+
+        protected virtual ISyntaxVisitorAction DefaultAction { get; }
+
+        public static ISyntaxVisitorAction Break { get; } = new BreakSyntaxVisitorAction();
+
+        public static ISyntaxVisitorAction Skip { get; } = new SkipSyntaxVisitorAction();
+
+        public static ISyntaxVisitorAction Continue { get; } = new ContinueSyntaxVisitorAction();
+
+        public ISyntaxVisitorAction Visit(
+            ISyntaxNode node,
+            TContext context)
+        {
+            // this instance has to go
```

Comment:
We should probably wrapp this with try and catch and just to be sure to no have a memory leak

Reply:
It is not that an issue if in error cases the pool loses one object ... it will than be garbage collection and the pool issues a new. In this case I think it would make the code to complicated.

Reply:
ok changed it ... it is now save ... the code still looks readable